tee-worker/litentry/core/credentials/src/jsonschema.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020-2023 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::achainable;
+use litentry_primitives::{AchainableParams, Assertion};
+use std::string::String;
+
+const BASE_URL: &str = "https://raw.githubusercontent.com/litentry/vc-jsonschema/main/dist/schemas";
+const NOT_IMPLEMENTED: &str =
+	"https://raw.githubusercontent.com/litentry/vc-jsonschema/main/dist/schemas/0-base.json";
+
+/// Returns the respective JSON Schema for the given assertion and its credential.
+/// JSON Schemas can be found on https://github.com/litentry/vc-jsonschema
+pub fn get_json_schema_url(assertion: Assertion, credential_subject_type: String) -> String {
+	match assertion {
+		Assertion::A1 => format!("{BASE_URL}/1-basic-identity-verification/1-0-0.json"),
+
+		Assertion::A2(_) => format!("{BASE_URL}/2-litentry-discord-member/1-0-0.json"),
+
+		Assertion::A3(_, _, _) => format!("{BASE_URL}/3-active-discord-id-hubber/1-0-0.json"),
+
+		Assertion::A4(_) => format!("{BASE_URL}/4-token-holding-time/1-0-0.json"),
+
+		Assertion::A6 => format!("{BASE_URL}/6-twitter-follower-amount/1-0-0.json"),
+
+		Assertion::A7(_) => format!("{BASE_URL}/4-token-holding-time/1-0-0.json"),
+
+		Assertion::A8(_) => format!("{BASE_URL}/7-evm-substrate-transaction-count/1-0-0.json"),
+
+		Assertion::A10(_) => format!("{BASE_URL}/4-token-holding-time/1-0-0.json"),
+
+		Assertion::A11(_) => format!("{BASE_URL}/4-token-holding-time/1-0-0.json"),
+
+		Assertion::A13(_) => format!("{BASE_URL}/8-decoded-2023-basic-special-badge/1-0-0.json"),
+
+		Assertion::A14 =>
+			format!("{BASE_URL}/9-polkadot-governance-participation-proof/1-0-0.json"),
+
+		Assertion::Achainable(params) => match params {
+			AchainableParams::AmountHolding(_) =>
+				format!("{BASE_URL}/19-token-holding-amount/1-0-0.json"),
+
+			AchainableParams::AmountToken(_) =>
+				format!("{BASE_URL}/19-token-holding-amount/1-0-0.json"),
+
+			// Generates both contract-creator and token-holder credentials.
+			AchainableParams::Amount(_) => {
+				if credential_subject_type == String::from("Contract Creator") {
+					return format!("{BASE_URL}/12-contract-creator/1-0-0.json")
+				} else if credential_subject_type == String::from("Token Holder") {
+					return format!("{BASE_URL}/11-token-holder/1-0-0.json")
+				}
```

Comment:
Can we eliminate the string comparison? Like have **one** JSON file for `AchainableParams::Amount` and use subfields for categorization?

Reply:
Sure, do you have a reference I could use?

Reply:
Wait, by one JSON you mean one schema? It will look a little off on the json-schema side due to the naming, but it should make this mapping easier.

Reply:
Done in https://github.com/litentry/litentry-parachain/pull/2473